### PR TITLE
SSG Test Suite: Fix (all) profile execution when running test suite in rule mode

### PR DESCRIPTION
**Pull request by ggbecker**
#### Description:

- Fix (all) profile execution when running test suite in rule mode.

### CODE REVIEW

File: tests/ssg_test_suite/rule.py [hunk 1]
```diff
@@ -18,7 +18,11 @@
 import data
 
 
-ALL_PROFILE_ID = "(all)"
+# Single quotes are needed because this value is used in
+# a bash environment when running oscap-ssh command line utility.
+# This has been fixed on OpenSCAP side but the fix has not been released yet:
+# https://github.com/OpenSCAP/openscap/pull/1366
```
Comment (matejak) [inline tests/ssg_test_suite/rule.py]: ditto to making this revert-friendly.

File: tests/ssg_test_suite/oscap.py [hunk 2]
```diff
@@ -117,7 +117,10 @@ def generate_fixes_remotely(formatting, verbose_path):
     ]
     command_operands = ['/{arf_file}'.format(** formatting)]
     if 'result_id' in formatting:
-        command_options.extend(['--result-id', formatting['result_id']])
+        # sanitize id because when using "(all)" profile the parenthesis
```
Comment (matejak) [inline tests/ssg_test_suite/oscap.py]: I propose to mark it somehow, so when a fix in `oscap-ssh` is released, these workarounds can be reverted quickly.

File: tests/ssg_test_suite/rule.py [hunk 3]
```diff
@@ -226,7 +230,10 @@ def _test_target(self, target):
 
     def _modify_parameters(self, script, params):
         if self.scenarios_profile:
-            params['profiles'] = [self.scenarios_profile]
```
Comment (matejak) [inline tests/ssg_test_suite/rule.py]: Let's tell users to specify the actual virtual profile name, i.e. `(all)`, like that, there won't be any need to treat the string `all` in a special way.

File: tests/ssg_test_suite/rule.py [hunk 4]
```diff
@@ -343,7 +350,8 @@ def perform_rule_check(options):
     checker.scenarios_profile = options.scenarios_profile
     # check if target is a complete profile ID, if not prepend profile prefix
     if (checker.scenarios_profile is not None and
-            not checker.scenarios_profile.startswith(OSCAP_PROFILE)):
+            not checker.scenarios_profile.startswith(OSCAP_PROFILE) and
+            not checker.scenarios_profile.lower() == "all"):
```
Comment (matejak) [inline tests/ssg_test_suite/rule.py]: Generally, we don't want to prefix anything to virtual profiles. A simple check could be based on brackets being present only in virtual profiles.

File: tests/ssg_test_suite/rule.py (hunk 1 above)
Comment (ggbecker) [reply tests/ssg_test_suite/rule.py]: done

File: tests/ssg_test_suite/rule.py (hunk 3 above)
Comment (ggbecker) [reply tests/ssg_test_suite/rule.py]: done

File: tests/ssg_test_suite/rule.py (hunk 4 above)
Comment (ggbecker) [reply tests/ssg_test_suite/rule.py]: done

File: tests/ssg_test_suite/oscap.py (hunk 2 above)
Comment (ggbecker) [reply tests/ssg_test_suite/oscap.py]: still needed. keeping the change in a nicer way.

**Comment by matejak** [comment]
Great, I can confirm that it works also on the system with quoting fixed in `oscap-ssh`.

**Pull request merged**